cmd/ingester/app/consumer/deadlock_detector.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumer
+
+import (
+	"runtime"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"github.com/uber/jaeger-lib/metrics"
+	"go.uber.org/zap"
+)
+
+// deadlockDetectorFactory is a factory for deadlockDetectors
+type deadlockDetectorFactory struct {
+	metricsFactory metrics.Factory
+	logger         *zap.Logger
+	interval       time.Duration
+	panicFunc      func(int32)
+}
+
+type deadlockDetector struct {
+	msgConsumed    *uint64
+	ticker         *time.Ticker
+	logger         *zap.Logger
+	closePartition chan struct{}
```

Comment:
I assume this does not apply to all-partitions detector?

Reply:
It does not